uber/decorators.py

```diff
@@ -110,6 +110,42 @@ def charge(self, session, payment_id, stripeToken, stripeEmail='ignored', **igno
     return charge
 
 
+def cached(func):
+    func.cached = True
+    return func
+
+
+def cached_page(func):
+    from sideboard.lib import config as sideboard_config
+    innermost = get_innermost(func)
+    func.lock = RLock()
+    @wraps(func)
+    def with_caching(*args, **kwargs):
+        if hasattr(innermost, 'cached'):
+            fpath = os.path.join(sideboard_config['root'], 'data', func.__module__ + '.' + func.__name__)
```

Comment:
The only page we're doing this with now is the schedule index, but theoretically we could apply this in other places.  In the long run though it's probably better to do this with nginx or something, but at least now we have the option.

So I figured it would be good to have a per-function lock.

The lock shouldn't slow things down in the expected case of a cache hit, and in the minority case of a cache miss, we don't want a bunch of page loads happening at the same time, since that would likely be slowing than everyone just waiting for the first one to complete.